src/js/profile/mobile/widget/Calendar.js

```diff
@@ -0,0 +1,301 @@
+				today = new Date(),
+				todayYear = today.getFullYear(),
+				todayMonth = today.getMonth() + 1,
+
+				Calendar = function () {
+					this.options = utilsObject.merge({}, defaultOptions);
+				},
+
+				defaultOptions = {
+					pastSelection: false
+				},
+
+				BaseWidget = ns.widget.BaseWidget,
+				prototype = new BaseWidget();
+
+			Calendar.prototype = prototype;
+
+			/**
+			* Init widget
+			* @method _init
+			* @param {HTMLElement} element
+			* @return {HTMLElement} Returns built element
+			* @member ns.widget.mobile.Calendar
+			* @protected
+			*/
+			prototype._init = function (element) {
+				var self = this;
+
+				if (!element.getAttribute("value")) {
+					element.setAttribute("value", self.options.value);
+				}
+
+				self._buildCalendar();
+
+				return element;
+			};
+
+			/**
+			* Draw a calendar on the table
+			* @method _buildCalendar
+			* @member ns.widget.mobile.Calendar
+			* @protected
+			*/
+			prototype._buildCalendar = function () {
+				var firstDate = new Date(todayYear, todayMonth - 1, 0),
+					lastDate = new Date(todayYear, todayMonth, 0),
+					day = firstDate.getDay(),
+					prevLastDate = lastDate.getDate(),
+					week = Math.ceil(prevLastDate / 7) + 1,
+					leftDays = 7,
+					setDays = 1,
+					nextMonthDate = 1,
+					cell = null,
+					calendar = document.getElementById("calendar"),
+					marginRow = calendar.insertRow();
+
+				marginRow.style.height = "7px";
+
+				for (var i = 1; i < week + 1; i++) {
+					var row = calendar.insertRow();
+
+					while (day != 0) { // prev disabled days.
+						day = day - 1;
+						leftDays = leftDays - 1;
+						cell = row.insertCell();
+						cell.innerHTML = prevLastDate - day;
+						cell.style.opacity = "10%";
+					}
+					while (leftDays != 0) {
+						if (setDays > prevLastDate) { // next disabled days.
+							cell = row.insertCell();
+							cell.innerHTML = nextMonthDate;
+							cell.style.opacity = "40%";
+							leftDays = leftDays - 1;
+							nextMonthDate = nextMonthDate + 1;
+						} else { // current enabled days.
+							cell = row.insertCell();
+							cell.innerHTML = setDays;
+							cell.style.opacity = "100%";
+							setDays = setDays + 1;
+							leftDays = leftDays - 1;
+						}
+					}
+					leftDays = 7;
+				}
+
+				document.getElementById("yearMonth").innerHTML = fullNameMonth[todayMonth - 1] + " " + todayYear;
+			}
+
+			/**
+			* Moving to another month erases the existing calendar
+			* @method _deleteCalendar
+			* @member ns.widget.mobile.Calendar
+			* @protected
+			*/
+			prototype._deleteCalendar = function () {
+				var calendar = document.getElementById("calendar");
+
+				while (calendar.rows.length > 2) {
+					calendar.deleteRow(2);
+				}
+			}
+
+			/**
+			* Click the arrow on the calendar to move to another month
+			* @method _onClick
+			* @param {HTMLElement} event
```

Comment:
event param is not HTMLElement.
Maybe it is Event.

Reply:
@param {Event} event
yes. you correct.

done.